File: 2019 maj/c++/2.cpp
#include <iostream>
#include <fstream>
using namespace std;

int silnia[10];

int rozbicie(int n)
{
    int ans = 0;
    while (n)
    {
        ans += silnia[n % 10];
        n /= 10;
    }
    return ans;
}

int main()
{
    ifstream in("liczby.txt");
    
    silnia[0] = 1;
    for (int i = 1; i <= 9; i++)
        silnia[i] = silnia[i - 1] * i;
    
    for (int i = 1; i <= 500; i++)
    {
        int a;
        in >> a;
        if (a == rozbicie(a))
            cout << a << '\n';
    }
}

File: 2019 maj/python/2.py
with open("liczby.txt", "r") as f:
    liczby = f.read().strip().split('\n')

silnia = [1] * 10
for i in range(1, 10):
    silnia[i] = silnia[i - 1] * i

for n in liczby:
    if int(n) == sum(silnia[int(i)] for i in list(n)):
        print(n)